### ISSUE TITLE
Cannot login with google or github when using @auth/core version 0.37.0 https://www.npmjs.com/package/@auth/core/v/0.37.0

### ISSUE THREAD

**Issue by ataponp** [get-convex/convex-auth#97]
Cannot login with google or github when upgrade @auth/core to version 0.37.0 
https://www.npmjs.com/package/@auth/core/v/0.37.0

TypeScript typecheck via `tsc` failed.
To ignore failing typecheck, use `--typecheck=disable`.
convex/auth.ts:17:30 - error TS2322: Type '(config: OAuthUserConfig<GitHubProfile> & { enterprise?: { baseUrl?: string | undefined; } | undefined; }) => OAuthConfig<GitHubProfile>' is not assignable to type 'AuthProviderConfig'.
  Type '(config: OAuthUserConfig<GitHubProfile> & { enterprise?: { baseUrl?: string | undefined; } | undefined; }) => OAuthConfig<GitHubProfile>' is not assignable to type '((...args: any) => (EmailConfig | WebAuthnConfig | OIDCConfig<any> | OAuth2Config<any> | CredentialsConfig<Record<string, CredentialInput>>) & InternalProviderOptions) & InternalProviderOptions'.
    Type '(config: OAuthUserConfig<GitHubProfile> & { enterprise?: { baseUrl?: string | undefined; } | undefined; }) => OAuthConfig<GitHubProfile>' is not assignable to type '(...args: any) => (EmailConfig | WebAuthnConfig | OIDCConfig<any> | OAuth2Config<any> | CredentialsConfig<Record<string, CredentialInput>>) & InternalProviderOptions'.
      Type 'OAuthConfig<GitHubProfile>' is not assignable to type '(EmailConfig | WebAuthnConfig | OIDCConfig<any> | OAuth2Config<any> | CredentialsConfig<Record<string, CredentialInput>>) & InternalProviderOptions'.  
        Type 'OIDCConfig<GitHubProfile>' is not assignable to type '(EmailConfig | WebAuthnConfig | OIDCConfig<any> | OAuth2Config<any> | CredentialsConfig<Record<string, CredentialInput>>) & InternalProviderOptions'. 
          Type 'OIDCConfig<GitHubProfile>' is not assignable to type 'OIDCConfig<any> & InternalProviderOptions'.
            Type 'OIDCConfig<GitHubProfile>' is not assignable to type 'OIDCConfig<any>'.
              Types of property 'client' are incompatible.
                Type 'Partial<Client & { token_endpoint_auth_method: string; }> | undefined' is not assignable to type 'Partial<Client> | undefined'.
                  Type 'Partial<Client & { token_endpoint_auth_method: string; }>' is not assignable to type 'Partial<Client>'.
                    Types of property 'token_endpoint_auth_method' are incompatible.
                      Type 'string | undefined' is not assignable to type 'ClientAuthenticationMethod | undefined'.
                        Type 'string' is not assignable to type 'ClientAuthenticationMethod | undefined'.    

17  providers: [CustomPassword, GitHub, Google],
                                ~~~~~~
convex/auth.ts:17:38 - error TS2322: Type '<P extends GoogleProfile>(options: OAuthUserConfig<P>) => OAuthConfig<P>' is not assignable to type 'AuthProviderConfig'.
  Type '<P extends GoogleProfile>(options: OAuthUserConfig<P>) => OAuthConfig<P>' is not assignable to type '((...args: any) => (EmailConfig | WebAuthnConfig | OIDCConfig<any> | OAuth2Config<any> | CredentialsConfig<Record<string, CredentialInput>>) & InternalProviderOptions) & InternalProviderOptions'.
    Type '<P extends GoogleProfile>(options: OAuthUserConfig<P>) => OAuthConfig<P>' is not assignable to type '(...args: any) => (EmailConfig | WebAuthnConfig | OIDCConfig<any> | OAuth2Config<any> | CredentialsConfig<Record<string, CredentialInput>>) & InternalProviderOptions'.
      Type 'OAuthConfig<any>' is not assignable to type '(EmailConfig | WebAuthnConfig | OIDCConfig<any> | OAuth2Config<any> | CredentialsConfig<Record<string, CredentialInput>>) & InternalProviderOptions'.
        Type 'OIDCConfig<any>' is not assignable to type '(EmailConfig | WebAuthnConfig | OIDCConfig<any> | OAuth2Config<any> | CredentialsConfig<Record<string, CredentialInput>>) & InternalProviderOptions'.
          Type 'OIDCConfig<any>' is not assignable to type 'OIDCConfig<any> & InternalProviderOptions'.      
            Type 'import("D:/Develop/BeDev/bedev-slack/node_modules/@auth/core/providers/oauth").OIDCConfig<any>' is not assignable to type 'import("D:/Develop/BeDev/bedev-slack/node_modules/@convex-dev/auth/node_modules/@auth/core/providers/oauth").OIDCConfig<any>'.
              Types of property 'client' are incompatible.
                Type 'Partial<Client & { token_endpoint_auth_method: string; }> | undefined' is not assignable to type 'Partial<Client> | undefined'.
                  Type 'Partial<Client & { token_endpoint_auth_method: string; }>' is not assignable to type 'Partial<Client>'.
                    Types of property 'token_endpoint_auth_method' are incompatible.
                      Type 'string | undefined' is not assignable to type 'ClientAuthenticationMethod | undefined'.
                        Type 'string' is not assignable to type 'ClientAuthenticationMethod | undefined'.    

17  providers: [CustomPassword, GitHub, Google],
<!--
Thanks for checking out Convex.

For help using Convex, check out our [Community Discord](https://convex.dev/community).
-->

**Comment by filigreti** [comment]
Im not able to log in with github, it shows the github prompt and redirects, but the user credentials don't get saved and the isAuthenticatedNextjs returns false

**Comment by dowski** [comment]
Hi, compatibility with `@auth/core` 0.37.x should land soon. It's checked in, just awaiting the next release.

**Comment by dowski** [comment]
This should be fixed now in versions >= 0.0.76.
